llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -537,59 +537,52 @@ void NVPTXAsmPrinter::emitKernelFunctionDirectives(const Function &F,
                                                    raw_ostream &O) const {
   // If the NVVM IR has some of reqntid* specified, then output
   // the reqntid directive, and set the unspecified ones to 1.
-  // If none of reqntid* is specified, don't output reqntid directive.
-  unsigned reqntidx, reqntidy, reqntidz;
-  bool specified = false;
-  if (!getReqNTIDx(F, reqntidx))
-    reqntidx = 1;
-  else
-    specified = true;
-  if (!getReqNTIDy(F, reqntidy))
-    reqntidy = 1;
-  else
-    specified = true;
-  if (!getReqNTIDz(F, reqntidz))
-    reqntidz = 1;
-  else
-    specified = true;
-
-  if (specified)
-    O << ".reqntid " << reqntidx << ", " << reqntidy << ", " << reqntidz
+  // If none of Reqntid* is specified, don't output reqntid directive.
+  unsigned Reqntidx, Reqntidy, Reqntidz;
+  Reqntidx = Reqntidy = Reqntidz = 1;
+  bool ReqSpecified = false;
+  if (getReqNTIDx(F, Reqntidx))
+    ReqSpecified |= true;
+  if (getReqNTIDy(F, Reqntidy))
+    ReqSpecified |= true;
+  if (getReqNTIDz(F, Reqntidz))
+    ReqSpecified |= true;
```

Comment:
I think I know what you were going for with those ors:
```suggestion
    ReqSpecified |= getReqNTIDx(F, Reqntidx);
    ReqSpecified |= getReqNTIDy(F, Reqntidy);
    ReqSpecified |= getReqNTIDz(F, Reqntidz);
```